### PR TITLE
add snax-strem-gemm-dev system

### PR DESCRIPTION
**Pull request by rgantonio**
This is a recreate of PR #31 since there was a synchronization issue. But this PR adds:

* Updated tests for the updated streamer and gemm components
* Adds the streamer and gemm generation
* A test to test the streamer gemm setup

### CODE REVIEW

File: .github/workflows/build-docker.yml
Comment (jorendumoulin) [inline .github/workflows/build-docker.yml:1]: undelete this file

**Comment by rgantonio** [comment]
It's clean!! Shall we push? 😄

**Review by jorendumoulin — APPROVED** [review]
Yes, good to go!

**Comment by rgantonio** [comment]
WAIT DONT SQUASH HOLD ON

**Comment by rgantonio** [comment]
OK now we Squash 😄 when the tests finish I will squash this!

**Pull request merged**